python/tvm/script/parse/tir/tir.py

```diff
@@ -80,6 +81,34 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
         self.eval_assign(target=lhs, source=rhs, bind_value=bind_value)
 
 
+@dispatch.register(token="tir", type_name="AugAssign")
+def visit_aug_ssign(self: Parser, node: doc.AugAssign) -> None:
+    new_rhs = deepcopy(node.target)
+    new_rhs.ctx = doc.Load(
+        node.value.lineno,
+        node.value.col_offset,
+        node.value.end_lineno,
+        node.value.end_col_offset,
+    )
+    new_node = doc.Assign(
+        [node.target],
+        doc.BinOp(
+            new_rhs,
+            node.op,
+            node.value,
+            node.value.lineno,
+            node.value.col_offset,
+            node.value.end_lineno,
+            node.value.end_col_offset,
+        ),
+        node.lineno,
+        node.col_offset,
+        node.lineno,
+        node.end_col_offset,
+    )
+    self.visit_Assign(new_node)
```

Comment:
btw this approach is cool if user inputs are correct; however, it won't produce proper results in error reporting as the column numbers are not accurate.

Instead, would you like to interpret the IR in the following workflow:
1. Evaluate RHS => a
2. Evaluate LHS with `ctx = doc.Load` => b
3. Evaluate `BinOp(npde.op, a, b)` => rhs
4. Evaluate LHS as normal assignment